ci-operator/step-registry/upi/install/powervs/cluster/upi-install-powervs-cluster-commands.sh

```diff
@@ -0,0 +1,406 @@
+  then
+    echo "ibmcloud CLI doesn't exist, installing"
+    curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
+  fi
+
+  retry "ic config --check-version=false"
+  retry "ic version"
+}
+
+function cleanup_ibmcloud_powervs() {
+  local version="${1}"
+  local workspace_name="${2}"
+
+  echo "Cleaning up prior runs - version: ${version} - workspace_name: ${workspace_name}"
+
+  echo "Cleaning up workspaces for ${workspace_name}"
+  for CRN in $(ic pi sl 2> /dev/null | grep "${workspace_name}" | awk '{print $1}' || true)
+  do
+    echo "Targetting power cloud instance"
+    retry "ic pi st ${CRN}"
+
+    echo "Deleting the PVM Instances"
+    for INSTANCE_ID in $(ic pi ins --json | jq -r '.pvmInstances[].pvmInstanceID')
+    do
+      echo "Deleting PVM Instance ${INSTANCE_ID}"
+      retry "ic pi ind ${INSTANCE_ID} --delete-data-volumes"
+      sleep 60
+    done
+
+    echo "Deleting the Images"
+    for IMAGE_ID in $(ic pi imgs --json | jq -r '.images[].imageID')
+    do
+      echo "Deleting Images ${IMAGE_ID}"
+      retry "ic pi image-delete ${IMAGE_ID}"
+      sleep 60
+    done
+
+    echo "Deleting the Network"
+    for NETWORK_ID in $(ic pi nets --json | jq -r '.networks[].networkID')
+    do
+      echo "Deleting network ${NETWORK_ID}"
+      retry "ic pi network-delete ${NETWORK_ID}"
+      sleep 60
+    done
+
+    retry "ic resource service-instance-update ${CRN} --allow-cleanup true"
+    sleep 30
+    retry "ic resource service-instance-delete ${CRN} --force --recursive"
+    for COUNT in $(seq 0 5)
+    do
+      FIND=$(ic pi sl 2> /dev/null| grep "${CRN}" || true)
+      echo "FIND: ${FIND}"
+      if [ -z "${FIND}" ]
+      then
+        echo "service-instance is deprovisioned"
+        break
+      fi
+      echo "waiting on service instance to deprovision ${COUNT}"
+      sleep 60
+    done
+    echo "Done Deleting the ${CRN}"
+  done
+
+  echo "Done cleaning up prior runs"
+}
+
+function create_powervs_service_instance() {
+  local version="${1}"
+  local workspace_name="${2}"
+  local powervs_zone="${3}"
+  local resource_group="${4}"
+
+  SERVICE_NAME=power-iaas
+  SERVICE_PLAN_NAME=power-virtual-server-group
+
+  # create workspace for powervs from cli
+  ic resource service-instance-create "${workspace_name}" "${SERVICE_NAME}" "${SERVICE_PLAN_NAME}" "${powervs_zone}" -g "${resource_group}" --allow-cleanup 2>&1 \
+  | tee /tmp/instance.id
+
+  # Process the CRN into a variable
+  CRN=$(cat /tmp/instance.id | grep crn | awk '{print $NF}')
+  export CRN
+  echo "${CRN}" > "${SHARED_DIR}"/POWERVS_SERVICE_CRN
+  POWERVS_SERVICE_INSTANCE_ID=$(echo "${CRN}" | sed 's|:| |g' | awk '{print $NF}')
+  export POWERVS_SERVICE_INSTANCE_ID
+  echo "${POWERVS_SERVICE_INSTANCE_ID}" > "${SHARED_DIR}"/POWERVS_SERVICE_INSTANCE_ID
+  sleep 30
+
+  # Tag the resource for easier deletion
+  ic resource tag-attach --tag-names mac-power-worker-"${CLEAN_VERSION}" --resource-id "${CRN}" --tag-type user
```

Comment:
Maybe this should come from an environment variable? tech debt? so we don't overwrite each other - phase1 / phase2 jobs